### ISSUE TITLE
Can I use same architecure picture between Transformer and LLaMA ?

### ISSUE THREAD

**Issue by jacksonchen1998** [facebookresearch/llama#346]
Hello everyone, I wonder that the main architecture between Transformer and LLaMA are the same, both are encoder-decoder model.

The main difference with the Transformer architecture:
- RMSNorm normalizing function
- The ReLU non-linearity is replaced by the SwiGLU activation function
- Absolute positional embeddings are removed and instead rotary positional embeddings (RoPE) are added at each layer of the network

Basically, I can use the picture from Transformer and editting three different parts, right ?

![img](https://pytorch.org/tutorials/_images/transformer_architecture.jpg)

Picture from [Language Modeling with nn.Transformer and torchtext — PyTorch](https://pytorch.org/tutorials/beginner/transformer_tutorial.html)

Thanks

**Comment by andrewPoulton** [comment]
The edits you've suggested are correct, but also Llama is a decoder-only model, so you'd have to remove the encoder part as well.  Also the norm functions come before each sublayer, with an additional norm occurring just before the output (it's a _pre_-norm architecture).

**Comment by jacksonchen1998** [comment]
@andrewPoulton OK. Thanks.
